docs: document SSH access for Jobs (#2557)
huggingface/hub-docs: docs/hub/jobs-configuration.md

['https://6a2ab384c4f53f9fc5aa4d4f--8000.hf.jobs']
```

## SSH

You can open an interactive SSH session into a running Job to debug, inspect, or work directly inside the container. Enable it at job creation with `--ssh` (CLI) or `ssh=True` (Python API), then connect with `hf jobs ssh <job_id>`.

> Only users with write access to the Job's namespace are allowed in (i.e. the Job creator, or members of the owner organization), authenticated by an SSH public key registered at [https://huggingface.co/settings/keys](https://huggingface.co/settings/keys).

Reviewer: ```suggestion
Only users with Write access to the Job's namespace are allowed (i.e. the Job creator, or members of the owner organization with Write permissions). Authentication is performed via SSH public keys registered at [https://huggingface.co/settings/keys](https://huggingface.co/settings/keys).
```

Reviewer: Perhaps we can also link to the `Namespace` section in the document so readers can understand how to run under an organization's namespace.

Author: I feel it's fine :shrug:
